### ISSUE TITLE
[BUG] `.to_gpu_matrix` function contains no implementation

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#2359]
**Describe the bug**
`.to_gpu_matrix` function contains no implementation . 

See:

https://github.com/rapidsai/cudf/blob/d3f19ca6f8047a0bb5fb70350b50d67dd53cacdc/python/cudf/cudf/dataframe/dataframe.py#L3053-L3063

**Steps/Code to reproduce bug**

```python
import cudf
df = cudf.DataFrame({'col_1':[1,2,3],'col_2':[3,4,5]})
out = df.to_gpu_matrix()
type(out)
```
Output:
```
NoneType
```

**Expected behavior**
To return a `numba gpu ndarray`  or raise a  `non-implemented error`.

**Comment by AK-ayush** [comment]
`to_gpu_matrix()` is redundant since `as_gpu_matrix()` is doing the same thing. So, I think we can scrape this method completely.

**Comment by brandon-b-miller** [comment]
closed by #2771 .
